MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,778 @@
+Interesting parties can subscribe to the event to get notified when the event is fired.
+The existing listener implementations for `SelectionService`, `GLSPCommandStack` & `EditorContextService` have been refactored to use the new API.
+Explicit methods for de-registering listeners are no longer available.
+Instead adopters have to keep track of the disposable that was received when registering the listener and dispose it when the listener is no longer needed.
+
+<details>
+  <summary>1.0.0</summary>
+
+```ts
+@injectable()
+export class MyService {
+    @inject(SelectionService)
+    protected selectionService: SelectionService;
+
+    protected listener: SelectionListener;
+
+    @postConstruct()
+    protected init(): void {
+        this.listener = {
+            selectionChanged: () => {
+                console.log('selection changed');
+            }
+        };
+        this.selectionService.register({
+            selectionChanged: () => {
+                console.log('selection changed');
+            }
+        });
+    }
+
+    dispose(): void {
+        this.selectionService.deregister(this.listener);
+    }
+}
+```
+
+</details>
+<details open>
+  <summary>2.x</summary>
+
+```ts
+@injectable()
+export class MyService implements Disposable {
+    @inject(SelectionService)
+    protected selectionService: SelectionService;
+
+    protected selectionListenerDisposable: Disposable;
+
+    @postConstruct()
+    protected init(): void {
+        this.selectionListenerDisposable = this.selectionService.onSelectionChanged(() => {
+            console.log('selection changed');
+        });
+    }
+
+    dispose(): void {
+        this.selectionListenerDisposable.dispose();
+    }
+}
+```
+
+<details>
+  <summary>List of changes</summary>
+
+-   SelectionListener -> ISelectionListener
+-   EditModeListener -> IEditModeListener
+
+</details>
+
+### Message API
+
+With 2.x the actions for sending information messages and/or status updates have been reworked.
+First of all they have been renamed
+
+-   ServerStatusAction -> StatusAction
+-   ServerMessageAction -> MessageAction
+
+Second, the `timeout` support for `MessageAction`'s has been removed.
+This feature was only well supported in the Theia integration and its main purpose was to dispatch progress information for long-running operations.
+A new dedicated API for progress reporting has been introduced which can be used for this purpose.
+If you have a long running information and want to report the progress you can use the following new actions:
+
+-   StartProgressAction
+-   UpdateProgressAction
+-   EndProgressAction
+
+</details>
+
+### Other breaking changes
+
+**IButtonHandler**
+
+Injecting an `IButtonHandler` constructor is now deprecated.
+Please use the `configureButtonHandler` method in your diagram DI module instead.
+
+**UndoRedo**
+
+`Undo` and `Redo` operations were incorrectly declared as actions in 1.0. With 2.x the have been migrated to operations.
```

Comment:
One more sentence for the Rename of the UndoRedoKeyListener?

Reply:
Why? It's just a basic rename, there is nothing more to it. Tbh I don`t exactly know why this was renamed in the first place.

Reply:
Yeah, I was wondering about that too. I just mentioned it because the sentence above just said we moved from actions to operations but nothing about the rename so it felt like sneaked into that list.